src/manifolds/SymplecticStiefel.jl

```diff
@@ -286,40 +286,27 @@ Return the total space of the [`SymplecticStiefel`](@ref) manifold, which is the
 get_total_space(::SymplecticStiefel{n,k,𝔽}) where {n,k,𝔽} = Symplectic{n,𝔽}()
 
 @doc raw"""
-    gradient(::SymplecticStiefel, f, p, backend::RiemannianProjectionBackend)
-    gradient!(::SymplecticStiefel, f, X, p, backend::RiemannianProjectionBackend)
-
-Compute the gradient of
-``f\colon \operatorname{SpSt}(2n, 2k) \rightarrow ℝ``
-at ``p \in \operatorname{SpSt}(2n, 2k)``.
-
-This function first computes the embedding gradient ``∇f(p) \in ℝ^{2n \times 2k}`` using
-the [`AbstractRiemannianDiffBackend`](@ref) in the [`RiemannianProjectionBackend`](@ref).
-Then it transforms the embedding gradient to the unique tangent vector space element
-``\text{grad}f(p) \in T_p\operatorname{SpSt}(2n, 2k)``
-which fulfills the variational equation
-````math
-    g_p(\text{grad}f(p), X)
-    = \text{D}f(p)
-    = \langle ∇f(p), X \rangle
-    \quad\forall\; X \in T_p\operatorname{SpSt}(2n, 2k).
-````
-The manifold gradient ``\text{grad}f(p)`` is computed from ``∇f(p)`` as
-````math
-    \text{grad}f(p) = ∇f(p)p^Tp + Q_{2n}p∇f(p)^TQ_{2n}p,
-````
+    X = riemannian_gradient(::SymplecticStiefel, f, p, Y)
+    riemannian_gradient!(::SymplecticStiefel, f, X, p, Y)
+
+Compute the riemannian gradient `X` of `f` on [`SymplecticStiefel`](@ref)  at a point `p`,
+provided that the gradient of the function ``\tilde f``, which is `f` continued into the embedding
+is given by `Y`.
+
+The manifold gradient `X` is computed from `Y` as
+```math
+    X = Yp^{\mathrm{T}}p + Q_{2n}pY^{\mathrm{T}}Q_{2n}p,
+```
 where ``Q_{2n}`` is the [`SymplecticMatrix`](@ref).
 """
-function gradient(::SymplecticStiefel, f, p, backend::RiemannianProjectionBackend)
-    amb_grad = _gradient(f, p, backend.diff_backend)
-    Q_p = SymplecticMatrix(p, amb_grad) * p
-    return amb_grad * (p' * p) .+ Q_p * (amb_grad' * Q_p)
+function riemannian_gradient(::SymplecticStiefel, p, Y)
+    Q_p = SymplecticMatrix(p, Y) * p
+    return Y * (p' * p) .+ Q_p * (Y' * Q_p)
 end
 
-function gradient!(::SymplecticStiefel, f, X, p, backend::RiemannianProjectionBackend)
-    _gradient!(f, X, p, backend.diff_backend)
-    Q_p = SymplecticMatrix(p, X) * p
-    X .= X * (p' * p) .+ Q_p * (X' * Q_p)
+function riemannian_gradient!(::SymplecticStiefel, X, p, Y)
```

Comment:
I think this function should accept the embedding metric kwarg if it's Euclidean.
```suggestion
function riemannian_gradient!(::SymplecticStiefel, X, p, Y; embedding_metric::EuclideanMetric=EuclideanMetric())
```

Reply:
I would prefer to just write `kwargs...` then?
Or maybe not – on the other hand, if the embedding metric is not Euclidean, this function would do something wrong, so it should maybe even error? I am not sure atm.

Reply:
Well, restricting it to `EuclideanMetric` removes the possibility of passing something else so it's safer though a bit more verbose.

Reply:
Hm, maybe I wrote that not precise enough.
Of you pass anything else _than_ EuclideanMetric_ this function does something _wrong_, but that is also because the embedding of Symplectic Stiefel is a euclidean space.

Reply:
Maybe we have to make the metric a positional argument (that is optional when calling it, defaults to `EuclideanMetric`) such that we can dispatch on that, too?

Reply:
Oh, indeed!

I am also not sure but if you have an embedding with a non-Euclidean metric, then we have that situation.

But thanks for the note, then this is ok :)